src/datumaro/plugins/missing_annotation_detection.py

```diff
@@ -0,0 +1,149 @@
+                return ""
+
+        label_categories: LabelCategories = self.categories()[AnnotationType.label]
+
+        class LabelGnosticMatcherContext(IMatcherContext):
+            def get_any_label_name(self, ann: Annotation, label_id: int) -> str:
+                return label_categories[label_id]
+
+        self._support_matchers = {
+            AnnotationType.bbox: BboxMatcher(
+                pairwise_dist=pairwise_dist,
+                context=LabelAgnosticMatcherContext()
+                if label_agnostic_matching
+                else LabelGnosticMatcherContext(),
+                match_segments=match_segments_more_than_pair,
+            ),
+        }
+
+    def _yield_item(
+        self, batch: List[DatasetItem], inference: List[List[Annotation]]
+    ) -> Generator[DatasetItem, None, None]:
+        for item, annotations in zip(batch, inference):
+            self._check_annotations(annotations)
+            yield self.wrap_item(
+                item,
+                annotations=self._find_missing_anns(
+                    gt_anns=item.annotations,
+                    pseudo_anns=self._apply_score_threshold(annotations),
+                ),
+            )
+
+    def _apply_score_threshold(self, annotations: List[Annotation]) -> List[Annotation]:
+        if self._score_threshold is None:
+            return annotations
+
+        return [
+            ann for ann in annotations if ann.attributes.get("score", 1.0) > self._score_threshold
+        ]
+
+    def _find_missing_anns(
+        self, gt_anns: List[Annotation], pseudo_anns: List[Annotation]
+    ) -> List[Annotation]:
+        for ann in pseudo_anns:
+            ann.attributes["_pseudo_label_"] = True
+
+        missing_labeled_anns = []
+        for ann_type, matcher in self._support_matchers.items():
+            clusters = matcher.match_annotations(
+                [
+                    [ann for ann in gt_anns if ann.type == ann_type],
+                    [ann for ann in pseudo_anns if ann.type == ann_type],
+                ]
+            )
+            for cluster in clusters:
+                ann = self._pick_missing_ann_from_cluster(cluster)
+                if ann is not None:
+                    missing_labeled_anns.append(ann)
+
+        return missing_labeled_anns
+
+    @staticmethod
+    def _pick_missing_ann_from_cluster(cluster: List[Annotation]) -> Optional[Annotation]:
+        pseudo_label_anns = []
+        gt_label_anns = []
+
+        for ann in cluster:
+            if "_pseudo_label_" in ann.attributes:
+                pseudo_label_anns.append(ann)
+            else:
+                gt_label_anns.append(ann)
+
+        if len(gt_label_anns) > 0:
+            return None
+
+        max_score = float("-inf")
+        max_ann = None
+        for ann in pseudo_label_anns:
+            score = ann.attributes.get("score", -1)
+            if score > max_score:
+                max_score = score
+                max_ann = ann
```

Comment:
Could you try with below?

```suggestion
        max_ann = max(pseudo_label_anns, key=lambda ann: ann.attributes.get("score", -1))
```

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/1049/commits/25a3817217c993a2f8cc46a90bf57a29639dfe72